SwiftCompilerSources/Sources/Optimizer/InstructionSimplification/SimplifyBeginAndLoadBorrow.swift

```diff
@@ -21,7 +21,52 @@ extension BeginBorrowInst : OnoneSimplifyable {
        !findPointerEscapingUse(of: borrowedValue)
     {
       tryReplaceBorrowWithOwnedOperand(beginBorrow: self, context)
+
+    } else if let thin2thickFn = borrowedValue as? ThinToThickFunctionInst,
+              uses.filterUsers(ofType: BranchInst.self).isEmpty
```

Comment:
Looks like `ThinToThickFunctionInst` always produces values of none ownership. I am not sure if the borrowedValue can ever be `ThinToThickFunctionInst `. Can you add a test and move this to a separate commit if this is intended ?